appcenter-data/android/src/main/java/com/microsoft/appcenter/reactnative/data/AppCenterReactNativeDataUtils.java

```diff
@@ -64,4 +69,67 @@ public static WritableArray convertJsonArrayToWritableArray(JsonArray jsonArray)
         }
         return writableArray;
     }
+
+    public static JsonObject convertReadableMapToJsonObject(ReadableMap readableMap) {
+        JsonObject jsonObject = new JsonObject();
+        if (readableMap == null) {
+            return jsonObject;
+        }
+        ReadableMapKeySetIterator iterator = readableMap.keySetIterator();
+        while (iterator.hasNextKey()) {
+            String key = iterator.nextKey();
+            ReadableType type = readableMap.getType(key);
+            switch (type) {
+                case Number:
+                    jsonObject.addProperty(key, readableMap.getDouble(key));
```

Comment:
I had an offline discussion with @jackycodes and @michaelsmitha on handling of different types of numbers (integer/long/double) here. 

Note that this line parse a JavaScript `number` value into Java `Double` value. **The conclusion is it is fine to do so because JSON doesn't care about integer/long/double types and treats them as JavaScript `number`.** Cosmos DB saves document as JSON. No matter it is integer `123`, long `123` or double `123.45`, it's treated as a JavaScript `number` in JSON.

An interesting edge case is `number` with trailing zeros (e.g., `123.00`) is considered as strictly equal (`===`) with the same number without trailing zeros (e.g., `123`). We verified Cosmos DB actually strips trailing zeros on saving the JSON document. In other words, the JSON documents in Cosmos would never have number properties with trailing zeros (e.g., `123.00`).

For example, `create` API scenarios like:
- JSON document property `123` would be saved as `123` in cosmos.
- JSON document property `123.45` would be saved as `123.45` in cosmos.
- JSON document property `123.00` would be saved as `123` in cosmos because of JSON not preserving trailing zeros.